tests/backwards_compatibility_tests/compatibility_test_runner.py

```diff
@@ -625,10 +717,23 @@ def copy_state_from_container(
 
     from_version = semver.VersionInfo.parse(args.from_version)
     to_version = semver.VersionInfo.parse(args.to_version)
+
+    # Basic validation that verifies: from_version shouldn't be greater than or equal to to_version
     if from_version >= to_version:
-        logger.debug("from_version should be less than to_version")
+        logger.error("from_version should be less than to_version")
         sys.exit(0) # TODO: figure out if we should just quit.
 
+    # Check for Major version compatibility
+    from_major_version = from_version.major
+    to_major_version = to_version.major
+
+    #If from major version & to major version aren't the same we cannot run backwards compatibility tests or rollback tests
+    if from_major_version != to_major_version:
+        logger.error(f"from Major version {from_major_version} & to Major version {to_major_version} cannot "
+                     f"be used for running backwards compatibility tests or rollback tests"
+                     f"since they are from different major versions")
+        sys.exit(0)
```

Comment:
when validation errors out, should we exit with an error code to indicate the test fails to run? Or maybe just throw an exception?

Reply:
oops, meant to use a sys.exit(1) here - thanks for catching this. I'll change it to sys.exit(1). I also explored the possibility of throwing an exception here, it will work but it may produce noisy logs which may include python stack trace etc. So I've made changes such that this will throw an exception and the exception will be caught, logged and sys.exit(1) will be called